torch/distributed/fsdp/flat_param.py

```diff
@@ -471,7 +471,12 @@ def _init_flat_param_and_metadata(
             raise ValueError(
                 f"Expects non-negative `aligned_numel` but got {aligned_numel}"
             )
-        dtype, requires_grad, device = self._validate_tensors_to_flatten(params)
+        (
+            dtype,
+            requires_grad,
+            device,
+            requires_grad_mask,
```

Comment:
maybe rename requires_grad and the mask to indicate one is for the entire flat param and the other is per param?